### PR TITLE
Add description field to nodes

### PR DESCRIPTION
**Pull request by stanjg**
Adds a description fields to nodes to identify them easier. For issue #936

### CODE REVIEW

File: database/migrations/2018_03_15_124536_add_description_to_nodes.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('nodes', function (Blueprint $table) {
+            $table->text('description')->nullable()->after('name');
```
Comment (DaneEveritt) [inline database/migrations/2018_03_15_124536_add_description_to_nodes.php]: You cannot have a `TEXT` field that is nullable in MySQL.

File: resources/themes/pterodactyl/admin/nodes/new.blade.php [hunk 2]
```diff
@@ -32,15 +32,16 @@
                         <input type="text" name="name" id="pName" class="form-control" value="{{ old('name') }}"/>
                         <p class="text-muted small">Character limits: <code>a-zA-Z0-9_.-</code> and <code>[Space]</code> (min 1, max 100 characters).</p>
                     </div>
+                    <div class="form-group">
+                        <label for="pDescription" class="form-label">Description</label>
+                        <textarea name="description" id="pDescription" rows="4" class="form-control" style="resize: vertical;">{{ old('description') }}</textarea>
+                        <p class="text-muted small">(max 2000 characters)</p>
```
Comment (DaneEveritt) [inline resources/themes/pterodactyl/admin/nodes/new.blade.php]: There should not be a limit on description size.

File: resources/themes/pterodactyl/admin/nodes/new.blade.php [hunk 3]
```diff
@@ -32,15 +32,16 @@
                         <input type="text" name="name" id="pName" class="form-control" value="{{ old('name') }}"/>
                         <p class="text-muted small">Character limits: <code>a-zA-Z0-9_.-</code> and <code>[Space]</code> (min 1, max 100 characters).</p>
                     </div>
+                    <div class="form-group">
+                        <label for="pDescription" class="form-label">Description</label>
+                        <textarea name="description" id="pDescription" rows="4" class="form-control" style="resize: vertical;">{{ old('description') }}</textarea>
```
Comment (DaneEveritt) [inline resources/themes/pterodactyl/admin/nodes/new.blade.php]: Please remove the `resize:vertical` to match the way the rest of the panel works.

File: app/Models/Node.php [hunk 4]
```diff
@@ -81,6 +82,7 @@ class Node extends Model implements CleansAttributes, ValidableContract
      */
     protected static $applicationRules = [
         'name' => 'required',
+        'description' => 'sometimes|max:2000',
```
Comment (DaneEveritt) [inline app/Models/Node.php]: This rule can simply be `string`

File: database/migrations/2018_03_15_124536_add_description_to_nodes.php (hunk 1 above)
Comment (stanjg) [reply database/migrations/2018_03_15_124536_add_description_to_nodes.php]: Oh, didn't know about that. Thanks.

**Comment by DaneEveritt** [comment]
Please include additions/modifications to the tests to cover cases where the description is passed in as a null value to ensure that it is properly converted to a string. I don't see any of that happening in the services currently.
